framework/src/userobject/ShapeElementUserObject.C

```diff
@@ -0,0 +1,32 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ShapeElementUserObject.h"
+
+template<>
+InputParameters validParams<ShapeElementUserObject>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  return params;
+}
+
+ShapeElementUserObject::ShapeElementUserObject(const InputParameters & parameters) :
+    ElementUserObject(parameters),
+    _phi(_assembly.phi()),
```

Comment:
This won't work... And demonstrates what I was worried about.

The issue is; what if this UserObject is coupled to multiple variables?  If one of those variables is linear Lagrange and the other is cubic hermite: then phi will be wrong.

To do this properly you're going to need to have an interface similar to Kernels and everything else (which is the right thing to do anyway) and have a special `computeElementUserObjectJacobianThread` that does the right thing.

I haven't yet thought of _how_ to store these Jacobians... I'll give it some more thought.